windows/uipriv_windows.hpp

```diff
@@ -1,4 +1,13 @@
 // 21 april 2016
+
+
+// the msys2 header has an unverified IDWriteTypography definition.
+// Would be good to get it verified and fixed in msys2, but in the meantime,
+// this define lets us compile....
+#if !defined(_MSC_VER)
+#define __MINGW_USE_BROKEN_INTERFACE
+#endif
+
```

Comment:
Another comment: this should go in winapi_windows.hpp, not uipriv_windows.hpp; this will help avoid build weirdnesses between files, should any happen to show up in the future...